net-utils/src/lib.rs

```diff
@@ -421,6 +426,10 @@ fn udp_socket_with_config(config: SocketConfig) -> io::Result<Socket> {
 
     let sock = Socket::new(Domain::IPV4, Type::DGRAM, None)?;
```

Comment:
I think some places in the code are bypassing this method; e.g.
https://github.com/anza-xyz/agave/blob/21d754000/core/src/banking_stage/forwarder.rs#L53
https://github.com/anza-xyz/agave/blob/21d754000/gossip/src/cluster_info.rs#L227
https://github.com/anza-xyz/agave/blob/21d754000/gossip/src/cluster_info.rs#L3005